pkg/generators/openapi.go

```diff
@@ -662,3 +674,95 @@ func (g openAPITypeWriter) generateSliceProperty(t *types.Type) error {
 	g.Do("},\n},\n},\n", nil)
 	return nil
 }
+
+// CustomArgs is used by the gengo framework to pass args specific to the API linter
+type CustomArgs struct {
+	ReportFilename string
+}
+
+type apiLinter struct {
+	rules          []APIRule
+	violations     []apiViolation
+	reportFilename string
+}
+
+func newAPILinter(reportFilename string) *apiLinter {
```

Comment:
what happens if the reportFilename == ""?

Reply:
I would like feedback about this: 

Current logic: if `reportFilename` is empty, the linter will glog any API rule violation and return **error** (the toolchain stops); if `reportFilename` is provided, the linter will print the violations to the report file and return **nil (no error)** (so the toolchain keeps going, assuming further tool will check the report file). This enables us to check in the known violations file in kubernetes, and then have Make rules to compare with the report file and make sure it doesn't get bigger.

Reply:
I recommend forcing users to choose either stdout (by passing "-", which can be the default) or a file name. In neither case would you use glog to print the errors. If they pass "", fail to construct.

Maybe stderr is a better place to write the errors than stdout.

Reply:
I agree with defaulting to "-" and printing the violations through stdout/stderr instead of glog. 

Does the idea of "returning nil when using report file, and returning error when using stdout/stderr" make sense to you? Alternatively we could always return error and let Make rules to catch the error when using report file.